### PR TITLE
Cherry pick details list filter icon PR to v7

### PR DESCRIPTION
**Pull request by khuynh92**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

#### Description of changes

- [x] cherry-pick changes from #18268 - DetailsList - add option to customize the filter icon
- [x] updated changes to the correct file path
#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by khuynh92** [comment]
Hey, @ecraig12345 - Here's the PR for the v7 change. Updated pr to have changes reflect in office-ui-fabric-react

File: change/office-ui-fabric-react-87af3408-8c21-4731-8137-0f19f77075e1.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "type": "minor",
+  "packageName": "office-ui-fabric-react",
```
Comment (ecraig12345) [inline change/office-ui-fabric-react-87af3408-8c21-4731-8137-0f19f77075e1.json:3]: You should add `"comment": "allow custom filter icons for the details header"` to this change file and delete the other one.

Comment (khuynh92) [reply change/office-ui-fabric-react-87af3408-8c21-4731-8137-0f19f77075e1.json:4]: Done!

**Comment by fabricteam** [comment]
## [Perf Analysis](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">7.0 Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/BaseButton-mount_base.html">961</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/BaseButton-mount.html">1007</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Breadcrumb-mount_base.html">41162</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Breadcrumb-mount.html">42242</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Checkbox-mount_base.html">1616</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Checkbox-mount.html">1645</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/CheckboxBase-mount_base.html">1363</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/CheckboxBase-mount.html">1373</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/ChoiceGroup-mount_base.html">5083</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/ChoiceGroup-mount.html">5058</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/ComboBox-mount_base.html">925</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/ComboBox-mount.html">937</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/CommandBar-mount_base.html">7704</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/CommandBar-mount.html">7628</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/ContextualMenu-mount_base.html">13155</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/ContextualMenu-mount.html">13246</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DefaultButton-mount_base.html">1166</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DefaultButton-mount.html">1213</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DetailsRow-mount_base.html">3786</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DetailsRow-mount.html">3767</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DetailsRowFast-mount_base.html">3829</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DetailsRowFast-mount.html">3826</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DetailsRowNoStyles-mount_base.html">3691</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DetailsRowNoStyles-mount.html">3586</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Dialog-mount_base.html">1722</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Dialog-mount.html">1757</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DocumentCardTitle-mount_base.html">1800</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/DocumentCardTitle-mount.html">1811</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Dropdown-mount_base.html">2571</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Dropdown-mount.html">2624</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/FocusTrapZone-mount_base.html">1695</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/FocusTrapZone-mount.html">1700</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/FocusZone-mount_base.html">1746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/FocusZone-mount.html">1775</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/IconButton-mount_base.html">1867</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/IconButton-mount.html">1818</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Label-mount_base.html">315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Label-mount.html">345</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Layer-mount_base.html">2002</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Layer-mount.html">2131</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Link-mount_base.html">448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Link-mount.html">453</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/MenuButton-mount_base.html">1527</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/MenuButton-mount.html">1546</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/MessageBar-mount_base.html">2009</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/MessageBar-mount.html">2111</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Nav-mount_base.html">3368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Nav-mount.html">3311</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/OverflowSet-mount_base.html">1407</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/OverflowSet-mount.html">1481</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Panel-mount_base.html">1466</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Panel-mount.html">1443</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Persona-mount_base.html">806</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Persona-mount.html">803</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Pivot-mount_base.html">1504</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Pivot-mount.html">1452</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/PrimaryButton-mount_base.html">1343</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/PrimaryButton-mount.html">1357</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Rating-mount_base.html">7988</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Rating-mount.html">8107</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SearchBox-mount_base.html">1336</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SearchBox-mount.html">1367</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Shimmer-mount_base.html">2753</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Shimmer-mount.html">2721</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Slider-mount_base.html">1560</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Slider-mount.html">1578</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SpinButton-mount_base.html">5284</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SpinButton-mount.html">5216</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Spinner-mount_base.html">419</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Spinner-mount.html">422</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SplitButton-mount_base.html">3319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SplitButton-mount.html">3245</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Stack-mount_base.html">492</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Stack-mount.html">523</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/StackWithIntrinsicChildren-mount_base.html">1653</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/StackWithIntrinsicChildren-mount.html">1613</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/StackWithTextChildren-mount_base.html">4954</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/StackWithTextChildren-mount.html">4991</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SwatchColorPicker-mount_base.html">10543</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/SwatchColorPicker-mount.html">10598</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/TagPicker-mount_base.html">2962</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/TagPicker-mount.html">2889</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/TeachingBubble-mount_base.html">49968</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/TeachingBubble-mount.html">50173</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Text-mount_base.html">440</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Text-mount.html">454</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/TextField-mount_base.html">1423</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/TextField-mount.html">1423</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Toggle-mount_base.html">870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/Toggle-mount.html">842</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/button-mount_base.html">103</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/18603/perf-test/button-mount.html">110</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Review by ecraig12345 — APPROVED** [review]

**Pull request merged**